### ISSUE TITLE
tape-eval vs. huggingface api? 

### ISSUE THREAD

**Issue by hyeh20** [songlab-cal/tape#82]
Hello, 
I was interested in evaluating the pretrained models in terms of stability/fluorescence without training the downstream model (just to gauge as a baseline). With the provided data, in the command line, I ran:

tape-eval transformer stability bert-base --metrics spearmanr

With the huggingface api, I ran:
`
model = ProteinBertForValuePrediction.from_pretrained('bert-base') 
`
and tokenizing with
`
tokenizer = TAPETokenizer(vocab='iupac')  
`
and proceeded further with tokenizing the sequences from the test set, passing them through the model and calculating spearmanr using scipy. 

I find that the spearmanr I achieve using each method is different with -0.139 in commandline while the spearmanr I get with the API vastly changes as I repeat trials. Is there a gap in my understanding or procedure on usage here?

Thanks!

**Comment by rmrao** [comment]
Hi - unfortunately we don't currently provide pretrained versions of the model for the downstream tasks (at least not in pytorch, the tensorflow models are available in the deprecated repo). You should be able to train one yourself using the framework (see here for hyperparameters that should work for fluorescence: https://app.wandb.ai/rmrao/fluorescence/reports/Fluorescence-Hyperparameter-Search--Vmlldzo5MjcwNA).

**Comment by hyeh20** [comment]
Thanks for the super quick reply! I was interested in just checking baseline metrics to see the model performance prior to downstream training (so a poor spearmanr is expected). I assumed that when I loaded in the ProteinBertForValuePrediction.from_pretrained('bert-base'), I would get the pretrained weights (from the language modeling, not trained from stability or fluorescence) and just a different last layer?

**Comment by rmrao** [comment]
Oh I see - sorry I misunderstood what you were doing. I think the spearmanr should change a fair amount with each trial, since the final layer will be randomly initialized. You could try explicitly setting some weights for the value prediction head, in which case you should always see the same result.

**Comment by hyeh20** [comment]
Thanks for the insight! It looks like when I run things in the commandline with:

tape-eval transformer stability bert-base --metrics spearmanr

I get the same spearmanr each time - is the initialization of the weights for the last layer here different from the api and not randomly initialized each trial?

**Comment by rmrao** [comment]
Yes, the tape API sets a random seed ([here](https://github.com/songlab-cal/tape/blob/8495202d78a5e0a7168702dffb512ce9b1fa7f58/tape/training.py#L560)) in order to get a reproducible result.

**Comment by rmrao** [comment]
You can pass in different arguments (I think `--seed` is the parameter) and you should see different results.

**Comment by hyeh20** [comment]
Thank you for the help, greatly appreciate it!
